### ISSUE TITLE
Question: How can someone use this for transaction emails?

### ISSUE THREAD

**Issue by patchthecode** [docker-mailserver/docker-mailserver#3615]
### Subject

I would like some feedback concerning a use case

### Description

This is a question (not a bug).
My app's user click on "Forgot Password".

I would like my app to talk to docker-mailserver to send an email to said user.  Does docker mail server have a solution for this? Or can it use some other project for this? I was thinking of using `Mautic`

**Comment by polarathene** [comment]
> I would like my app to talk to docker-mailserver to send an email to said user.

- Submit mail to submission ports (465 or 587) with an authenticated user (`setup email add ...`).
- The authenticated username does not have anything to do with the sender address (_except if `SPOOF_PROTECTION=1` is enabled_). So you can use whatever sender address you like (eg: `noreply@example.com`).
- App does the same thing, use email client/library to create an email and send to a submission port with the required credentials for sending.

After you handle that, the mail will be sent outbound on port 25 to be delivered to the recipient address (destination).

Usually to avoid issues with sender reputation you'll relay your mail through a service like SendGrid.
- In that case you configure DMS to relay outbound mail through the smarthost/relayhost, which may send mail to service like SendGrid on a different port (_non-standard mail ports they offer, or standard submission ports_).
- You could also avoid DMS entirely, and just send mail via their API, or submitting the mail to SendGrid. If you don't intend for DMS to receive mail to store and retrieve, then a send-only DMS is referred to as a null-mailer.

If you don't use services like SendGrid, your mail may not be delivered or more likely to arrive in spam/junk for the user. Be advised that many VPS prevent sending mail out on port 25, so you may need to rely on a relay service regardless.

---

**TL;DR:**
- DMS provides no API.
- DMS can send the mail, but you must submit it as mail to DMS through a mail client / library.
